lib/charms/loki_k8s/v0/loki.py

```diff
@@ -90,6 +270,84 @@ def unit_ip(self) -> str:
             return str(bind_address)
         return ""
 
+    def _remove_alert_rules_files(self, container) -> None:
+        """Remove alert rules files"""
+
+        for f in container.list_files(RULES_DIR):
```

Comment:
As discussed on Mattermost this iteration to remove each file is only necessary because you are using a volume mount. So I presume you have made an explicit decision to do so. Is there a specific reason for this ? Not data is already persisted in the relation.